plugins/outputs/health/health.go

```diff
@@ -143,7 +148,13 @@ func (h *Health) listen() (net.Listener, error) {
 
 func (h *Health) ServeHTTP(rw http.ResponseWriter, _ *http.Request) {
 	var code = http.StatusOK
-	if !h.isHealthy() {
+
+	healthy := h.isHealthy()
+	if h.UnhealthyIfMetricsAreDelayed && h.firstMetricReceived {
+		healthy = healthy && time.Since(h.lastMetricTime) < time.Duration(h.MaxTimeBetweenMetrics)
+	}
```

Comment:
```suggestion
	if h.MaxTimeBetweenMetrics > 0 {
		healthy = healthy && time.Since(h.lastMetricTime) < time.Duration(h.MaxTimeBetweenMetrics)
	}
```

Reply:
Hmmm shouldn't we rather set the `h.lastMetricTime` to `now` in `Connect` as otherwise the output stays healthy if no metric arrives at all...

Reply:
I want the output to be healthy if no metric is received. This is because I want to avoid the case where telegraf is restarted (because of being unhealthy) on startup.

Reply:
with the last change, `lastMetricTime` is set to `now` in `Connect`.